### PR TITLE
Vrf impl

### PR DESCRIPTION
**Pull request by AOranov**

### CODE REVIEW

File: spectrum-vrf/src/tests.rs [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+            vrf_gen::<Secp256k1>().unwrap();
+
+        let m_hash: Sha2Digest256 = sha256_hash("Test_tx".as_bytes());
+
+        let proof =
+            vrf_prove::<Secp256k1>(vrf_sk, m_hash).unwrap();
+        let valid =
+            vrf_verify::<Secp256k1>(vrf_pk, m_hash.clone(), proof).unwrap();
+
+        assert_eq!(valid, true);
```
Comment (oskin1) [inline spectrum-vrf/src/tests.rs]:
```suggestion
        assert!(valid);
```
You can use assert! simply to test booleans

File: spectrum-vrf/src/lottery.rs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    let c_fb: FieldBytes<TCurve> = proof.c.into();
+    let s_fb: FieldBytes<TCurve> = proof.s.into();
+
+    let proof_bytes = [projective_point_to_bytes::<TCurve>(proof.gamma),
+        c_fb.to_vec(), s_fb.to_vec()].concat();
+    let proof_hash = blake2b256_hash(&proof_bytes);
+    let proof_hash_bytes: [u8; 32] = proof_hash.into();
+    let proof_hash_bytes_vec = proof_hash_bytes.to_vec();
+
+    let proof_num = u64::from_ne_bytes(proof_hash_bytes_vec[0..8].try_into().unwrap());
```
Comment (oskin1) [inline spectrum-vrf/src/lottery.rs]: Are you sure its safe to project u256 to u64?

Comment (AOranov) [reply spectrum-vrf/src/lottery.rs]: This is exact the last question that I want to resolve, I didn't found distinct from_bytes methods for u256 in the elliptic_curve lib

Comment (oskin1) [reply spectrum-vrf/src/lottery.rs]: https://docs.rs/num-bigint/latest/num_bigint/struct.BigUint.html

**Review by oskin1 — APPROVED** [review]

**Pull request merged**